S3/input.tf

```diff
@@ -0,0 +1,61 @@
+###
+# Common tags
+###
+variable "billing_tag_key" {
+  description = "(optional) The name of the billing tag"
+  type        = string
+  default     = "CostCentre"
+}
+
+variable "billing_tag_value" {
+  description = "(required) The value of the billing tag"
+  type        = string
+}
+
+variable "bucket_name" {
+  description = "(Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name."
+  type        = string
+  default     = null
+}
+
+variable "bucket_prefix" {
+  description = "(Optional, Forces new resource) Creates a unique bucket name beginning with the specified prefix. Conflicts with bucket."
+  type        = string
+  default     = null
+}
+
+variable "tags" {
+  description = "(Optional) A mapping of tags to assign to the bucket."
+  type        = map(string)
+  default     = {}
+}
+
+variable "force_destroy" {
+  description = "(Optional, Default:false ) A boolean that indicates all objects should be deleted from the bucket so that the bucket can be destroyed without error. These objects are not recoverable."
+  type        = bool
+  default     = false
+}
+
+variable "versioning" {
+  description = "Map containing versioning configuration."
```

Comment:
For consistency, how about an `(Optional)` or `(Required)` prefix to all var descriptions.